scripts/check-asciidoctor-build.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+set -e
+
+# ensure asciidoctor is installed
+if ! command -v asciidoctor ;
+then
+    echo "Asciidoctor is not installed. Please install it and try again."
+    exit 1
+fi
+
+# list of *.adoc modules in the pull request
+MODULES=$(git diff-tree HEAD HEAD~1 --no-commit-id --name-only -r | awk '/modules\/(.*)\.adoc/')
+# list of *.adoc assemblies in the pull request
+PAGES=$(git diff-tree HEAD HEAD~1 --no-commit-id --name-only -r | awk '!/modules\/(.*)\.adoc/' | awk '/(.*)\.adoc/')
+
+if [ -n "${MODULES}" ] ;
+then
+    echo "Building assemblies that include updated modules. Build will fail with FAILED, ERROR, or WARNING messages..."
+    for MODULE in $MODULES
+        do UPDATED_PAGES=$(find . -name "*.adoc" -type f -exec grep -l "$MODULE" {} \; | sed 's|^\./||' | sed 's|\.adoc||' )
+    done
+
+    for PAGE in $UPDATED_PAGES
+        do asciidoctor "$PAGE" -a icons! -o /dev/null | grep -Ei 'asciidoctor: (FAILED|ERROR|WARNING)' && echo "Build errors found" && exit 1
```

Comment:
This could be made into a function so that it can be reused for both modules and assemblies. That way, if we need to change the command being run, we only have to change it in one location.

I also think we might be better off using the `--failure-level` flag and checking for a non-zero exit code rather than searching for a specific string. This is just in case Asciidoctor starts to use a different output format some day (unlikely, but you never know), and it should also run a bit faster. :slightly_smiling_face:

Reply:
I think we could also potentially pass the entire list of pages directly to the `asciidoctor` command instead of running it once for each page in a `for` loop, but I'm not sure if that would change the logic in any way. I don't _think_ so, but can't check it myself at the moment. :slightly_smiling_face:

Reply:
Sounds good! Will do.

Reply:
OK, took your suggestions on board. `--failure-level` fails early and fast. First error it finds the build will bomb. This is better, though it doesn't report all the errors in the PR.